src/main/java/com/google/javascript/gents/RemoveGoogScopePass.java

```diff
@@ -0,0 +1,69 @@
+package com.google.javascript.gents;
+
+import com.google.javascript.jscomp.AbstractCompiler;
+import com.google.javascript.jscomp.CompilerPass;
+import com.google.javascript.jscomp.NodeTraversal;
+import com.google.javascript.rhino.Node;
+import com.google.javascript.rhino.Token;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+
+/**
+ * Compiler pass removing the {@code goog.scope} wrapper, so that the module and class
+ * transformations can take effect.
+ */
+public final class RemoveGoogScopePass extends AbstractTopLevelCallback implements CompilerPass {
+
+  private final AbstractCompiler compiler;
+
+  public RemoveGoogScopePass(AbstractCompiler compiler) {
+    this.compiler = compiler;
+  }
+
+  @Override
+  public void process(Node externs, Node root) {
+    NodeTraversal.traverseEs6(compiler, root, this);
+  }
+
+  @Override
+  public void visit(NodeTraversal t, Node n, Node parent) {
+    Node maybeCallNode = n.getFirstChild();
+    if (maybeCallNode == null || maybeCallNode.getToken() != Token.CALL) {
+      return;
+    }
+
+    String callName = maybeCallNode.getFirstChild().getQualifiedName();
+    if (!"goog.scope".equals(callName)) {
+      return;
+    }
+
+    // Extract the goog.scope contents, and add them to module being constructed.
+    Node blockOfScopeContents = n.getLastChild().getLastChild().getLastChild();
+    blockOfScopeContents.detachFromParent();
+
+    // Collect all of the children into a list before detaching them from the parent,
+    // since detaching breaks the "getNext()" chain.
+    List<Node> scopeChildren = new ArrayList<>();
+    @Nullable Node maybeChildToAdd = blockOfScopeContents.getFirstChild();
+    while (maybeChildToAdd != null) {
+      scopeChildren.add(maybeChildToAdd);
```

Comment:
optional: I think you can collapse these two loops, by always taking the first child here (both in the while condition and the body), and then removing that child and appending it at the insertion point.